src/controller.cpp

```diff
@@ -249,6 +249,40 @@ void controllerT::execute_command_internal(std::string command, size_t counter,
 	worker_counter_cv.notify_all();
 }
 
+controllerT::execute_config controllerT::generate_config(const size_t requested_cpus, const resource_selector selector) const {
```

Comment:
To be honest, I am not a fan of this solution. From my understanding this is part of the scheduler logic. The scheduler selects the required resources. The controller just manages the running instances and keep track of which job is running at which node/slot.

Maybe we should move the later part to the scheduler instead of moving scheduler logic here?